CRM/Event/BAO/Event.php

```diff
@@ -140,6 +140,17 @@ public static function create(&$params) {
       }
       $params['created_date'] = date('YmdHis');
     }
+    // CRM-18169 Check for start date change
+    if (!empty($params['id']) && CRM_Contribute_PseudoConstant::checkContributeSettings('deferred_revenue_enabled')) {
+      $prevEvent = new CRM_Event_DAO_Event();
+
+      $prevEvent->id = $params['id'];
+      $prevEvent->find(TRUE);
+      if (date('m', strtotime($params['start_date']) != date('m')) &&
+        (date('Ymd', strtotime($prevEvent->start_date)) != date('Ymd', strtotime($params['start_date'])))) {
+        self::updateDeferredTransactions($params);
```

Comment:
The solution would be to maintain a cron job of some sort which do the updating of transactions in batches. Since changing the start date of an event is an edge case, I don't know if we should support this right now.
@JoeMurray thoughts?